### ISSUE TITLE
[D214, D405, D417] False positive with arg keyword `attributes`

### ISSUE THREAD

**Issue by JP-Ellis** [astral-sh/ruff#16007]
## Summary

If using Google style and putting the argument's description on a new line, then if one argument is `attributes`, it raises D214 and D405.

I raised a similar issue previously (#8457), and this appears to another related edge case.

## Example

<details>
<summary><code>ruff.toml</code></summary>

```toml
[lint.pydocstyle]
convention = "google"
```

</details> 

```python
def send(payload: str, attributes: dict[str, Any]) -> None:
    """
    Send a message.

    Args:
        payload:
            The message payload.

        attributes:
            Additional attributes to be sent alongside the message.
    """
```

```console
❯ ruff check --config config.toml --select ALL ruff.py 
ruff.py:1:5: D417 Missing argument description in the docstring for `send`: `attributes`
  |
1 | def send(payload: str, attributes: dict[str, Any]) -> None:
  |     ^^^^ D417
2 |     """
3 |     Send a message.
  |
ruff.py:9:9: D405 [*] Section name should be properly capitalized ("attributes")
   |
 7 |             The message payload.
 8 |
 9 |         attributes:
   |         ^^^^^^^^^^ D405
10 |             Additional attributes to be sent alongside the message.
11 |     """
   |
   = help: Capitalize "attributes"

ruff.py:9:9: D214 [*] Section is over-indented ("attributes")
   |
 7 |             The message payload.
 8 |
 9 |         attributes:
   |         ^^^^^^^^^^ D214
10 |             Additional attributes to be sent alongside the message.
11 |     """
   |
   = help: Remove over-indentation from "attributes"

Found 6 errors.
```

_Note that I omitted a couple of expected errors (such as an undefined 'Any')_

<details>
<summary><code>ruff --version</code></summary>

```
ruff 0.9.3
```

</details>

## Related

This is related (nearly identical) to #8457.

**Comment by ntBre** [comment]
I realized I never responded here, but thank you for the report! This looks like a bug, and I have a PR open to address it.
